pkg/controller/pxc/version.go

```diff
@@ -0,0 +1,98 @@
+package pxc
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+
+	api "github.com/percona/percona-xtradb-cluster-operator/pkg/apis/pxc/v1"
+	"github.com/percona/percona-xtradb-cluster-operator/pkg/pxc/app/statefulset"
+	"github.com/robfig/cron"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const jobName = "ensure-version"
+
+func (r *ReconcilePerconaXtraDBCluster) deleteEnsureVersion(id int) {
+	r.crons.crons.Remove(cron.EntryID(id))
+	delete(r.crons.jobs, jobName)
+}
+
+func (r *ReconcilePerconaXtraDBCluster) ensurePXCVersion(cr *api.PerconaXtraDBCluster, vs VersionService, sfs *statefulset.Node) error {
+	shedule, ok := r.crons.jobs[jobName]
+
+	if ok && cr.Spec.UpgradeOptions.Schedule == "" {
+		r.deleteEnsureVersion(shedule.ID)
+		return nil
+	}
+
+	if ok && shedule.CronShedule == cr.Spec.UpgradeOptions.Schedule {
+		return nil
+	}
+
+	if ok {
+		log.Info(fmt.Sprintf("remove job %s because of new %s", shedule.CronShedule, cr.Spec.UpgradeOptions.Schedule))
+		r.deleteEnsureVersion(shedule.ID)
+	}
+
+	log.Info(fmt.Sprintf("add new job: %s", cr.Spec.UpgradeOptions.Schedule))
+	id, err := r.crons.crons.AddFunc(cr.Spec.UpgradeOptions.Schedule, func() {
```

Comment:
can we create a regular (non-lambda) function and call it when the user creates a new cluster? to avoid the situation when the newly created cluster requires update